main.c

```diff
@@ -181,6 +188,90 @@ static void show_table_dump(ryzen_access ry, int any_adjust_applied)
 	//don't free current_table_values because this would deinitialize our table
 }
 
+// Check if launched from ryzen-controller
+static int check_parent() {
+#ifndef _WIN32
+	pid_t ppid = getppid();
+	char pathname[512];
+	char buf[512];
+	char gt[38] = "/opt/Ryzen Controller/ryzen-controller";
+	int cx;
+	cx = snprintf(pathname, 512, "/proc/%d/exe", ppid);
+
+	if (cx >= 0 && cx < 512) {
+		ssize_t l = readlink(pathname, buf, 512);
+		if (l > 0) {
+			if (strncmp(buf, gt, 38) == 0) {
+				return 1;
+			}
+		}
+	}
+#endif
+	return 0;
+}
+
+// Are we launched via a root user?
+static int is_user_root() {
+#ifndef _WIN32
+	uid_t uid = getuid();
+	return uid == 0;
+#else
+	return 1;
+#endif
+}
+
+// Are we in ryzenadj group?
+static int is_in_ryzenadj_group() {
+#ifndef _WIN32
+	int ngroups = 0;
+	uid_t uid;
+	struct passwd *pw;
+	struct group *gr_allow;
+	gr_allow = getgrnam("ryzenadj");
```

Comment:
I see that you did add the group documentation to the readme. Would be nice to mention here that we expect that the sudo assignment is done in this group because it took me some time to understand it.

Reply:
Can add an extra comment in the src.